src/request.rs

```diff
@@ -690,3 +690,17 @@ fn no_hostname() {
     );
     assert!(req.get_host().is_err());
 }
+
+#[test]
+fn request_implements_send_and_sync() {
+    let _request: Box<dyn Send> = Box::new(Request::new(
+        &Agent::default(),
+        "GET".to_string(),
+        "https://example.com/".to_string(),
+    ));
+    let _request: Box<dyn Sync> = Box::new(Request::new(
+        &Agent::default(),
+        "GET".to_string(),
+        "https://example.com/".to_string(),
+    ));
```

Comment:
I've learnt a better pattern for this using an empty body generic `fn`.

```rust
  fn ensure_send(_v: impl Send) {}
  fn ensure_sync(_v: impl Sync) {}

#[test]
 fn request_implements_send_and_sync() {
    ensure_send(Request::new(…));
    ensure_sync(Request::new(…));
 }
```

Reply:
Ooh, that's a nice technique!